src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1597,11 +1598,12 @@ def _ssl_context():
     return ssl.create_default_context()
 
 
-def _build_app_settings_output(app_settings, slot_cfg_names):
+def _build_app_settings_output(app_settings, slot_cfg_names, show_values=True):
     slot_cfg_names = slot_cfg_names or []
+    app_settings = _mask_creds_related_appsettings(app_settings)
     return [{'name': p,
              'value': app_settings[p],
-             'slotSetting': p in slot_cfg_names} for p in _mask_creds_related_appsettings(app_settings)]
+             'slotSetting': p in slot_cfg_names} for p in (app_settings if show_values else _redact_appsettings(app_settings))]
```

Comment:
We should maintain the old behavior if show_values is set to true. In this case, we should still call _mask_creds_related_appsettings

Reply:
I added line 1603 appsettings = _mask_creds_related_appsettings, which should overshadow (and hopefully make it easier to delete that once it gets fixed and is no longer necessary)

Reply:
`_mask_creds_related_appsettings` only masks the value of `DOCKER_REGISTRY_SERVER_PASSWORD`:

https://github.com/Azure/azure-cli/blob/3024c5c86f22f9a0f6cc7def626d458d06d62267/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L1808-L1811

https://github.com/Azure/azure-cli/blob/3024c5c86f22f9a0f6cc7def626d458d06d62267/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L1696

Masking `DOCKER_REGISTRY_SERVER_PASSWORD` may not be necessary if the user explicitly asks to show values.